### PR TITLE
MBS-12266 Share module dependencies logic

### PR DESCRIPTION
**Pull request by eugene-krivobokov**
I need to reuse the logic of finding dependencies in a module.
Implementation in `module-dependencies-graph` plugin looks good enough to reuse.
I've extracted it to a shared module.

### CODE REVIEW

File: subprojects/gradle/impact-shared/build.gradle.kts [hunk 1]
```diff
@@ -5,7 +5,7 @@ plugins {
 }
 
 dependencies {
-    api(projects.subprojects.gradle.moduleDependenciesGraph)
```
Comment (eugene-krivobokov) [inline subprojects/gradle/impact-shared/build.gradle.kts:8]: We consumed here a plugin that is worse.

File: subprojects/gradle/module-dependencies-shared/src/main/kotlin/com/avito/module/dependencies/ModuleConfigurationDependencies.kt [hunk 2]
```diff
@@ -4,20 +4,21 @@ import com.avito.module.configurations.ConfigurationType
 import org.gradle.api.Project
 import org.gradle.api.artifacts.ProjectDependency
 
-public fun Project.dependenciesOnProjects(
-    types: Set<Class<out ConfigurationType>>
-): Set<ProjectDependency> {
+public fun Project.directDependenciesOnProjects(
+    types: Set<ConfigurationType>
+): Map<ConfigurationType, Set<Project>> {
```
Comment (eugene-krivobokov) [inline subprojects/gradle/module-dependencies-shared/src/main/kotlin/com/avito/module/dependencies/ModuleConfigurationDependencies.kt:9]: We lost a mapping before. 
This is a client's responsibility to decide what to do with this information.

File: subprojects/gradle/module-dependencies-shared/src/main/kotlin/com/avito/module/configurations/ConfigurationType.kt [hunk 3]
```diff
@@ -2,31 +2,31 @@ package com.avito.module.configurations
 
 import org.gradle.api.artifacts.Configuration
 
-public sealed class ConfigurationType {
+public enum class ConfigurationType {
```
Comment (eugene-krivobokov) [inline subprojects/gradle/module-dependencies-shared/src/main/kotlin/com/avito/module/configurations/ConfigurationType.kt:5]: Don't see the benefits of using a sealed class here.
Clients have to convert all the time to class.

**Review by dsvoronin — APPROVED** [review]

File: settings.gradle.kts [hunk 4]
```diff
@@ -38,6 +38,7 @@ include(":subprojects:gradle:graphite-config")
 include(":subprojects:gradle:impact")
 include(":subprojects:gradle:impact-shared")
 include(":subprojects:gradle:module-dependencies-graph")
+include(":subprojects:gradle:module-dependencies-shared")
```
Comment (sergioboi) [inline settings.gradle.kts:41]: I'm not confident in that name. It would be good If you could replace `shared` with something meaningful

Comment (eugene-krivobokov) [reply settings.gradle.kts:41]: Agree, I'll try to find more obvious name

Comment (eugene-krivobokov) [reply settings.gradle.kts:41]: #1298

**Pull request merged**